packages/material-ui/src/Link/Link.js

```diff
@@ -0,0 +1,184 @@
+// @inheritedComponent ButtonBase
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { componentPropType } from '@material-ui/utils';
+import ButtonBase from '../ButtonBase';
+import { fade } from '@material-ui/core/styles/colorManipulator';
+import classNames from 'classnames';
+import { capitalize } from '@material-ui/core/utils/helpers';
+import withStyles from '@material-ui/core/styles/withStyles';
+
+export const styles = theme => ({
+  /* Styles applied to the root element. */
+  root: {
+    lineHeight: 1.75, // To remove with v4.
+    ...theme.typography.button,
+    textTransform: null,
+    boxSizing: 'border-box',
+    minWidth: 64,
+    padding: '6px 16px',
+    borderRadius: theme.shape.borderRadius,
+    color: theme.palette.text.primary,
+    transition: theme.transitions.create(['background-color', 'box-shadow', 'border'], {
+      duration: theme.transitions.duration.short,
+    }),
+    '&:hover': {
+      textDecoration: 'none',
+      backgroundColor: fade(theme.palette.text.primary, theme.palette.action.hoverOpacity),
+      // Reset on touch devices, it doesn't add specificity
+      '@media (hover: none)': {
+        backgroundColor: 'transparent',
+      },
+      '&$disabled': {
+        backgroundColor: 'transparent',
+      },
+    },
+    '&$disabled': {
+      color: theme.palette.action.disabled,
+    },
+  },
+  /* Styles applied to the span element that wraps the children. */
+  label: {
+    width: '100%', // assure the correct width for iOS Safari
+    display: 'inherit',
+    alignItems: 'inherit',
+    justifyContent: 'inherit',
+  },
+  /* Styles applied to the root element if `color="primary"`. */
+  primary: {
+    color: theme.palette.primary.main,
+    '&:hover': {
+      backgroundColor: fade(theme.palette.primary.main, theme.palette.action.hoverOpacity),
+      // Reset on touch devices, it doesn't add specificity
+      '@media (hover: none)': {
+        backgroundColor: 'transparent',
+      },
+    },
+  },
+  /* Styles applied to the root element if `color="secondary"`. */
+  secondary: {
+    color: theme.palette.secondary.main,
+    '&:hover': {
+      backgroundColor: fade(theme.palette.secondary.main, theme.palette.action.hoverOpacity),
+      // Reset on touch devices, it doesn't add specificity
+      '@media (hover: none)': {
+        backgroundColor: 'transparent',
+      },
+    },
+  },
+  /* Styles applied to the root element if `color="inherit"`. */
+  colorInherit: {
```

Comment:
👍

Reply:
We can put the hover the effect in a style rule we don't apply when `color="inherit"`?